### PR TITLE
[WIP] [Composer] Enhancement: Clean up helpers/bin/run

### PR DESCRIPTION
**Pull request by localheinz**
This PR

* [ ] cleans up `composer/helpers/bin/run`

💁‍♂ Let me know if you'd prefer me to squash the commits - I've tried to make small, reviewable changes for ease of review.

### CODE REVIEW

**Comment by localheinz** [comment]
Rebased after merge of #1332.

**Comment by localheinz** [comment]
@greysteil 

Please let me know if you'd prefer me to squash the commits after review and before merge!

File: composer/helpers/bin/run [hunk 1]
```diff
@@ -5,78 +5,95 @@ declare(strict_types=1);
...
+if ($memoryLimit !== '-1' && $memoryInBytes($memoryLimit) < 1024 * 1024 * 1900) {
+    @ini_set('memory_limit', '1900M');
 }
 
-date_default_timezone_set('Europe/London');
+// Set user defined memory limit
+$memoryLimit = getenv('COMPOSER_MEMORY_LIMIT');
 
-// This storage is freed on error (case of allowed memory exhausted)
-$memory = str_repeat('*', 1024 * 1024);
```
Comment (greysteil) [inline composer/helpers/bin/run]: Looks like you're removing this, but I think it's important. Previously we were seeing OOM errors fail to get written - allocating some memory here that can be freed up when processing them fixes that.

Comment (localheinz) [reply composer/helpers/bin/run]: Removed the commit!

**Pull request closed without merging**